cli/src/program_v4.rs

```diff
@@ -111,52 +108,44 @@ impl ProgramV4SubCommands for App<'_, '_> {
                 .setting(AppSettings::SubcommandRequiredElseHelp)
                 .subcommand(
                     SubCommand::with_name("deploy")
-                        .about("Deploy a program")
+                        .about("Deploy a new or redeploy an existing program")
                         .arg(
-                            Arg::with_name("program_location")
+                            Arg::with_name("program-location")
                                 .index(1)
                                 .value_name("PROGRAM_FILEPATH")
                                 .takes_value(true)
                                 .help("/path/to/program.so"),
                         )
                         .arg(
-                            Arg::with_name("program")
-                                .long("program")
-                                .value_name("PROGRAM_SIGNER")
+                            Arg::with_name("start-offset")
+                                .long("start-offset")
```

Comment:
Are `start-offset` and `end-offset` only valid when `buffer` is being used? Or, can these be used with the final `program` account? If it's the former, we can add `conficts_with()` here.

Reply:
It is independent. One can do partial deployments with and without using a buffer account.